### ISSUE TITLE
Version 0.7 -- Can't find the buttons

### ISSUE THREAD

**Issue by mvdgb** [OsaSoft/youtube-better-subscriptions#41]
Using Firefox 61.0.2 (64 bit) on Win 7, with YBS 0.6.1, and looking at my YouTube subscriptions page, the buttons (Hide All As Watched and Hide As Watched) are in the upper right of the Today section. With YBS 0.7, I can't find them anywhere. I've had to drop back to 0.6.1.

**Comment by LimEJET** [comment]
Confirmed, 0.7 seems broken. 0.6.1 still works. Firefox 61.0.2 64 bit, Win 10

**Comment by mvdgb** [comment]
I understand that they moved with the latest update, but I can't locate them on the top bar or anywhere else now.

![image](https://user-images.githubusercontent.com/36159719/44293894-62e69a80-a255-11e8-8202-fb37209100fc.png)

**Comment by LimEJET** [comment]
@macearl That is Youtube's default layout in Firefox. Don't ask me why it looks like that, but I started my browser in safe mode just to be sure, and it looks the same.

For me with 0.7, watched videos are no longer hidden, and no buttons are visible. I personally don't care about the buttons, but the videos not being hidden are a problem.

[For reference, this is the source of a watched thumbnail item](https://gist.github.com/LimEJET/ee813047631a6bc590754192df713081) in my browser.

**Comment by OsaSoft** [comment]
I havent figured out a way to get the old layout in subs feed, but I managed to get when going to https://www.youtube.com/account
Ill use that to see where the buttons need to be added.

@LimEJET I imagine the videos arent getting hidden since adding the buttons throws an exception and so the rest of the script fails to execute. Ill have to add a failsafe for cases like these...

Im currently out of town, so wont be able to do any fixes until sunday or monday

**Comment by LimEJET** [comment]
Since 0.6.1 still works, I'd say this is a regression.

**Comment by macearl** [comment]
Going to: https://www.youtube.com/feed/subscriptions?disable_polymer=1 seems to still work, if not try the steps here: https://www.reddit.com/r/youtube/comments/82wjto/youtube_removes_the_way_to_restore_the_old/dvg9tw7

It seems to fail at https://github.com/OsaSoft/youtube-better-subscriptions/blob/f076ac04266a0229e9050b58c8234f48e9bf9807/pageHandler.js#L23-L29

`pageloader` will be null after line 24 and therefore the script will be in an endless loop

I also downgraded to 0.6.1 and to me it looks like there are other things broken as well, for me the mark as watched buttons do not line up correctly at all and when the sites tries to load more videos there are some null exceptions
![oldlayout](https://user-images.githubusercontent.com/1804541/44300066-624f1200-a2f0-11e8-9e8d-2011141b0b07.png)
![nullerror](https://user-images.githubusercontent.com/1804541/44300065-624f1200-a2f0-11e8-8818-10c2bb8d5be0.png)

@LimEJET i can't test this on my site as the new design is the default for me but could you try using:
https://www.youtube.com/feed/subscriptions?disable_polymer=0 and see if you get the new design this way? probably not but might be worth a shot

Ideally youtube would switch everyone over to the "new" layout that way there would only be one that needs to be maintained

**Comment by LimEJET** [comment]
That flag randomly makes the site flip randomly between Polymer and the old UI.

There is a deeper issue here: Youtube does user agent sniffing to serve different content to browsers other than Chrome because only Chrome implements nonstandard API's used by the site. [Documented here](https://twitter.com/cpeterso/status/1021626510296285185). The result is a 5x slowdown in non-Chrome browsers when running the Polymer UI.

I have an extension that resets the disable_polymer flag and forces the old UI. It probably stuffs something in a cookie, which is why the old UI still showed up in Safe Mode.

I get that you don't want to have to maintain changes for two UI's. As 0.6.1 works for me, it's not really an issue in this case. However, as long as Polymer is slower on other browsers than Chrome, I'm betting you'll keep having this issue pop up.

**Comment by OsaSoft** [comment]
I see. I do use FF as my primary browser and havent really had any issues with Polymer, so never noticed that.
However, as long as there are not too many versions of the ui, Id like to keep support for all of them.
Thanks for bringing this up

**Comment by EddyGurge** [comment]
For me, I just went to https://www.youtube.com/new and clicked the "Get it Now" and it updated to the latest YouTube layout, and everything started working again just fine.

**Comment by dmodave** [comment]
How does one even revert back to 0.6.1? I don't know how to do this. I'm on Mac, with FF 61.0.2

**Comment by andrewmyles** [comment]
Yeah, switching to the new layout works, but you shouldn't have moved the buttons. I shouldn't have to switch the layout just for this addon to work.

**Comment by macearl** [comment]
Please keep in mind that you are using a free addon, developed by a single individual in his free time and that the buttons had to be moved to work consistently while navigating youtube.

The developer also has already stated that he will take a look at it as soon as he can and intends to keep supporting the old layout.

Now for everyone who does not want to switch to the new layout just yet (im guessing youtube will completely disable the old one soon anyway) all you need to do is downgrade the addon to an older version for that go to:
https://addons.mozilla.org/en-US/firefox/addon/youtube-better-subscriptions/versions/

and install an older version, e.g. 0.6.1

then go to your list of installed addons click more on "Better Subscriptions for YouTube" and disable automatic updates

For chrome i am not aware of an official way to download older versions of an addon and the only two sites which host a mirror of old versions https://crx.dam.io/ and https://www.crx4chrome.com/ do not seem to have this one. So you would need to download an older version from the source code here: https://github.com/OsaSoft/youtube-better-subscriptions/archive/df9833bb2751c0735202ccf48286281e9288253a.zip

extract that then open the extension list in chrome remove the already installed version, enable the developer mode click on "Load unpacked" and pick the unpacked folder, now you should have the old version, to get back to the newest version remove the old version and install it again from the webstore
